### ISSUE TITLE
SSR error in useWindowScrollPosition and useOnWindowScroll 

### ISSUE THREAD

**Issue by jekh** [imbhargav5/rooks#1135]
**Describe the bug**
`ReferenceError: window is not defined` in `useWindowScrollPosition` on 5.11.7.

**This causes SSR pages to break completely in next.js**, and probably other SSR frameworks, too.

**To Reproduce**
Use `useWindowScrollPosition` in an SSR-rendered page:
```ts
  const { scrollY } = useWindowScrollPosition()
```

**Expected behavior**
Since `window` is undefined, `scrollY`, `scrollX`, etc. should probably be 0, or the return value of the function should be undefined.

**Screenshots**
Stack trace:
```
error - ReferenceError: window is not defined
    at useOnWindowScroll (/code/node_modules/.pnpm/rooks@5.11.7_sfoxds7t5ydpegc3knd667wn6m/node_modules/rooks/dist/cjs/hooks/useOnWindowScroll.js:17:70)
    at useWindowScrollPosition (/code/node_modules/.pnpm/rooks@5.11.7_sfoxds7t5ydpegc3knd667wn6m/node_modules/rooks/dist/cjs/hooks/useWindowScrollPosition.js:33:47)
```

The offending code seems to be here: https://github.com/imbhargav5/rooks/blob/main/src/hooks/useOnWindowScroll.ts#L18

I think it probably just needs a simple `if (typeof window === "undefined")` guard

**Additional context**
I'm using next.js 12.2, but I would expect this error to occur in any SSR context.

It looks like this error is related to the change here: https://github.com/imbhargav5/rooks/pull/1030 , which removed the `typeof window === "undefined"` guards without removing references to `window` itself.

**Comment by imbhargav5** [comment]
Fixed!

**Comment by sgehrman** [comment]
fixed?  doesn't work for me.  I'm on the latest build.

**Comment by dgknca** [comment]
I get this error for `useDimensionsRef`

**Comment by imbhargav5** [comment]
These are hooks which require the window object and aren't useful with SSR. Will not be fixed.

**Comment by dgknca** [comment]
> These are hooks which require the window object and aren't useful with SSR. Will not be fixed.

How can I make this hook client-only? Here is my component in NextJS:

```js
function HeaderRoot({ children }: { children?: ReactNode }) {
  const [headerContext, setHeaderContext] = useContext(HeaderContext);
  const { showHeader } = headerContext;
  const { scrollY } = useWindowScrollPosition();
  const [ref, headerDimensions] = useDimensionsRef();
```

**Comment by imbhargav5** [comment]
@dgknca So the correct fix as mentioned in the next.js docs will be to move the extract a component which requires the useWindowScrollPosition and useDimensionsRef and then encapsulate them with @next/dynamic

https://nextjs.org/docs/advanced-features/dynamic-import#with-no-ssr
```jsx
import dynamic from 'next/dynamic'

const DynamicClientHeader = dynamic(() => import('../components/client-component'), {
  ssr: false,
})
```

<DynamicClientHeader/>

**Comment by equinusocio** [comment]
> @dgknca So the correct fix as mentioned in the next.js docs will be to move the extract a component which requires the useWindowScrollPosition and useDimensionsRef and then encapsulate them with @next/dynamic
> 
> https://nextjs.org/docs/advanced-features/dynamic-import#with-no-ssr
> 
> ```js
> import dynamic from 'next/dynamic'
> 
> const DynamicClientHeader = dynamic(() => import('../components/client-component'), {
>   ssr: false,
> })
> ```

This is not a fix, not everyone is using Next.js, and adding another dep just to avoid a bug coming from this lib is dumb. The hook `useDimensionRef` is still bugged and it needs to check if `window` exist.

**Comment by imbhargav5** [comment]
>This is not a fix, not everyone is using Next.js, and adding another dep just to avoid a bug coming from this lib is dumb. The hook useDimensionRef is still bugged and it needs to check if window exist since is not client-only

@equinusocio  What are you on about? This is a fix for Next.js, which this issue is about (original context), and my reply is also to a user who asked for Next.js. It's dumb (your great word) that you missed the original context that was mentioned in the issue. 
Also next/dynamic is not even another dependency. It is included within Next.js. Pretty genius of you to not know that but calling other solutions dumb. It's literally present for `This is useful if an external dependency or component relies on browser APIs like window`. Their words not mine (see link). 

I am not fixing a window related hook for SSR at this point. It's not my priority because it makes no sense to me to use it on the server. If I eventually fix it, this thread will be notified.

**Comment by equinusocio** [comment]
My gosh. You have to learn a lot about oss. You just closed my original issue with a meh comment. 

You suggested to install react-no-ssr which IS another dep. Soo, yes, to use this hook outside next we need to add another dep. I don't even care about next or dynamic since is not what i'm talking about. 

We learned you don't care to have a community around oss since you closed the other issue without even trying to understand it. 

To recap, you don't want to fix this with a check on window to make it client-only (for real) while other hooks already have such check. My god.

**Comment by imbhargav5** [comment]
> We learned you don't care to have a community around this lib since you closed the other issue without even trying to understand it.

No I closed the issue because I labelled it as `wont-fix`. In other words, I have no intention of fixing it.

> To recap, you don't want to fix this with a check on window to make it client-only (for real) while other hooks already have such check.

Yep. Once again, for the love of god. No intention of fixing it. And good luck using window dimensions on the server. 

> Don't try to fool me 😉

You dont need my help there. You have it covered.
